scripts/switchBranches.fish

```diff
@@ -1,5 +1,25 @@
 #!/usr/bin/env fish
 
+function checkoutRepo
+  if test (count $argv) -ne 2
+      echo "Checkout needs two parameters branch force"
+      return 1
+  end
+  set -l branch $argv[1]
+  set -l clean $argv[2]
+
+  git checkout -- .
+  and git fetch
+  and git checkout $branch
+  and if test "$clean" = "true"
+    git reset --hard origin/$branch
```

Comment:
will this work if the branch name contains whitespace?

Reply:
can we trim branch names before using them ?